src/main/java/de/tum/in/www1/artemis/service/ConsistencyCheckService.java

```diff
@@ -0,0 +1,62 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.service.dto.ConsistencyErrorDTO;
+import de.tum.in.www1.artemis.service.programming.ProgrammingExerciseService;
+
+/**
+ * Service Implementation for consistency checks
+ * of programming exercises
+ */
+@Service
+public class ConsistencyCheckService {
+
+    private final Logger log = LoggerFactory.getLogger(ConsistencyCheckService.class);
+
+    private final ProgrammingExerciseService programmingExerciseService;
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    public ConsistencyCheckService(ProgrammingExerciseService programmingExerciseService, ProgrammingExerciseRepository programmingExerciseRepository) {
+        this.programmingExerciseService = programmingExerciseService;
+        this.programmingExerciseRepository = programmingExerciseRepository;
+    }
+
+    /**
+     * Check VCS and CI consistency of a given programming exercise
+     *
+     * @param exerciseId of the programming exercise to check
+     * @return List containing the resulting errors, if any.
+     */
+    public List<ConsistencyErrorDTO> checkConsistencyOfProgrammingExercise(long exerciseId) {
```

Comment:
The reason why I included the implementation in the `ProgrammingExerciseService` was because of the Service dependencies to VCS and CI which could be reused. 

As I look at it now, I can confirm your remark and will move the code to `ConsistencyCheckService`